From 0c35c7663d21ffab65bb5a12ea262b3c0f8d13d0 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Mon, 23 Oct 2017 11:06:12 +0800
Subject: [PATCH 2694/5242] MLK-16694-1: ASoC: wm8960: add pm runtime for
 wm8960

commit  7455c7ef445acba8b3c0b3d0c2196c84e1c29e27 from
https://source.codeaurora.org/external/imx/linux-imx.git

As in imx8 mek board, the codec's mclk is from the audio subsystem,
the subsystem's power domain should be disabled when subsystem is idle.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/wm8960.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/codecs/wm8960.c b/sound/soc/codecs/wm8960.c
index f425e88..c17c5357 100644
--- a/sound/soc/codecs/wm8960.c
+++ b/sound/soc/codecs/wm8960.c
@@ -18,6 +18,7 @@
 #include <linux/clk.h>
 #include <linux/i2c.h>
 #include <linux/slab.h>
+#include <linux/pm_runtime.h>
 #include <sound/core.h>
 #include <sound/pcm.h>
 #include <sound/pcm_params.h>
@@ -1431,6 +1432,8 @@ static int wm8960_i2c_probe(struct i2c_client *i2c,
 
 	i2c_set_clientdata(i2c, wm8960);
 
+	pm_runtime_enable(&i2c->dev);
+
 	ret = devm_snd_soc_register_component(&i2c->dev,
 			&soc_component_dev_wm8960, &wm8960_dai, 1);
 
-- 
1.7.9.5

